File: my app/src/musicSearch.js
async function searchItunes(query, limit = 5) {
  const q = String(query || "").trim();
  if (!q) return [];

  const capped = Math.max(1, Math.min(Number(limit) || 5, 10));
  const url = new URL("https://itunes.apple.com/search");
  url.searchParams.set("term", q);
  url.searchParams.set("entity", "song");
  url.searchParams.set("limit", String(capped));

  const res = await fetch(url, { method: "GET" });
  if (!res.ok) return [];

  const data = await res.json().catch(() => null);
  const results = data?.results;
  if (!Array.isArray(results)) return [];
  return results.filter((r) => r && typeof r === "object");
}

module.exports = { searchItunes };

File: my app/music_search.py
from __future__ import annotations

from typing import Any

import aiohttp


async def search_itunes(query: str, limit: int = 5) -> list[dict[str, Any]]:
    """
    iTunes Search API orqali trek qidiradi.
    Bu API natijada previewUrl (30s) va trackViewUrl linklarni berishi mumkin.
    """
    q = (query or "").strip()
    if not q:
        return []

    params = {
        "term": q,
        "entity": "song",
        "limit": str(max(1, min(limit, 10))),
    }

    async with aiohttp.ClientSession() as session:
        async with session.get("https://itunes.apple.com/search", params=params, timeout=20) as resp:
            if resp.status != 200:
                return []
            data = await resp.json()
            results = data.get("results") or []
            return [r for r in results if isinstance(r, dict)]

